server/src/main/java/org/elasticsearch/index/mapper/RangeFieldMapper.java

```diff
@@ -145,12 +145,14 @@ public void locale(Locale locale) {
         @Override
         protected void setupFieldType(BuilderContext context) {
             super.setupFieldType(context);
-            DateFormatter dateTimeFormatter = fieldType().dateTimeFormatter;
+            DateFormatter formatter = fieldType().dateTimeFormatter;
             if (fieldType().rangeType == RangeType.DATE) {
-                if (!locale.equals(dateTimeFormatter.locale())) {
-                    fieldType().setDateTimeFormatter(dateTimeFormatter.withLocale(locale));
+                if (Strings.hasLength(builder.pattern) &&
```

Comment:
just the matter of style I guess, but the expression here ended up pretty long
we might as well extract and call it hasPatternChanged ?

Reply:
done